paraphrase-job.sh

```diff
@@ -2,60 +2,86 @@
 
 . /opt/genie-toolkit/lib.sh
 
-parse_args "$0" "train_or_gen owner dataset_owner task_name experiment dataset model" "$@"
+parse_args "$0" "owner dataset_owner project experiment \
+            input_dataset output_dataset filtering_model paraphrasing_model_full_path skip_generation" "$@"
 shift $n
 
 set -e
 set -x
 
-aws s3 sync s3://almond-research/${dataset_owner}/dataset/${experiment}/${dataset} dataset/
-
-modeldir="$HOME/models/$model"
-mkdir -p "$modeldir"
-rm -fr "$modeldir/dataset"
-mkdir "$modeldir/dataset"
-rm -fr "$modeldir/cache"
-mkdir -p "$modeldir/cache"
-ln -s "$HOME/dataset" "$modeldir/dataset/${task_name}"
-ln -s $modeldir /home/genie-toolkit/current
-mkdir -p "/shared/tensorboard/${experiment}/${owner}/${model}"
-
-
-if [ "$train_or_gen" = "train" ]
-then
-  # run paraphrase training script
-  genienlp train-paraphrase \
-    --train_data_file dataset/para_freeform_train.txt \
-    --eval_data_file dataset/para_freeform_dev.txt \
-    --output_dir "$modeldir" \
-    --tensorboard_dir "/shared/tensorboard/${experiment}/${owner}/${model}" \
-    --model_type gpt2 \
-    --do_train \
-    --do_eval \
-    --evaluate_during_training \
-    --overwrite_output_dir \
-    --logging_steps 1000 \
-    --save_steps 1000 \
-    --max_steps 40000 \
-    --save_total_limit 1 \
-    "$@"
-  aws s3 sync $modeldir/ s3://almond-research/${owner}/models/${experiment}/${model}
+aws s3 sync s3://almond-research/${dataset_owner}/dataset/${project}/${experiment}/${input_dataset} ${input_dataset}/
+aws s3 sync s3://almond-research/${owner}/models/${project}/${experiment}/${filtering_model} ${filtering_model}/
+aws s3 sync s3://almond-research/${paraphrasing_model_full_path} paraphraser/
+
+mkdir -p ${output_dataset}
+cp -r ${input_dataset}/* ${output_dataset}
+
+on_error () {
+  # on failure upload the outputs to S3
+  aws s3 sync ${output_dataset} s3://almond-research/${dataset_owner}/dataset/${project}/${experiment}/${output_dataset}
+}
```

Comment:
I don't particularly like this approach of shipping the dataset to s3 on failure, because datasets and temporary files accumulate and we pay storage costs for nothing.

Reply:
The idea was that if generation succeeds but filtering fails, that generation is still useful and we can just run the filtering again. Also, even without failures, I upload 3 intermediate files (`paraphrased.tsv`, `unfiltered.tsv`, `filtered.tsv`) to S3 as they can be used in different experiments. I will change this to only the final file and `unfiltered.tsv` after paraphrasing configurations are finalized.

Reply:
I don't mind uploading intermediate files on success. It's on failure that I'm worried. In other scripts, on failure it goes to a separate directory, which nobody ever looks into. Here I see it goes into the normal directory, which is kind of weird...